src/orchestrator-process/src/lib.rs

```diff
@@ -525,6 +527,19 @@ impl NamespacedProcessOrchestrator {
                 }
             }
 
+            // Adding scope guard to clean up scratch directory
```

Comment:
```suggestion
            // Clean up scratch directory when the service is terminated.
            // This is best effort as a development and testing convenience.
            // Because the process orchestrator is not used in production, we 
            // don't need to be perfectly robust with the cleanup.
```